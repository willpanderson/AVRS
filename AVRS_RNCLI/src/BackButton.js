import React from 'react';
import PropTypes from 'prop-types';

import { Pressable, Text, View, StyleSheet } from 'react-native';

class BackButton extends React.Component {

    static propTypes = {
        backPress: PropTypes.func.isRequired,
    }

    render() {
        return (
            <View>
                <Pressable 
                    style={({ pressed }) => [
                        styles.button,
                        pressed ? { opacity: 0.8 } : {},
                    ]}
                    onPress={this.props.backPress}>
                    <Text>Back</Text>
                </Pressable>
            </View>
    );
  }
}

const styles = StyleSheet.create({
    button: {
        width: 50,
        backgroundColor: '#377888',
    },

})

export default BackButton;
