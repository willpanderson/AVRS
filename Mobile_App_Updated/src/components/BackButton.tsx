import {Pressable, StyleSheet, View} from 'react-native';
import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';
// import Icon from '';

const BackButton = (props: {
  backPress: (value?: React.SetStateAction<string>) => void;
  args?: string;
}) => {
  return (
    <View>
      <Pressable
        style={({pressed}) => [styles.button, pressed ? {opacity: 0.8} : {}]}
        onPress={() => props.backPress(props.args)}>
        <Icon name="chevron-back" size={35} />
      </Pressable>
    </View>
  );
};

export default BackButton;

const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
  },
});
