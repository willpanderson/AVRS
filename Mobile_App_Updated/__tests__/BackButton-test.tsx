import 'react-native';
import * as React from 'react';
import BackButton from '../src/components/BackButton';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';

describe('Tests for BackButton component', () => {
  it('Renders correctly', () => {
    const tree = renderer.create(
      <BackButton backPress={() => console.log('hello')} />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Click button calls passed function', () => {
    const setInputType = jest.fn();
    const {getByText} = render(<BackButton backPress={setInputType} />);

    fireEvent.press(getByText('Back'));
    expect(setInputType).toBeCalled();
  });
});
