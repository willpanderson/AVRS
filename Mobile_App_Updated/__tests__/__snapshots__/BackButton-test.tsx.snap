// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for BackButton component Renders correctly 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#377888",
          "width": 50,
        },
        Object {},
      ]
    }
  >
    <Text>
      Back
    </Text>
  </View>
</View>
`;
